File: TestTask3/src/scripts/comment.ts
export default class CommentBox {
  comment: Comment
  commentBox: HTMLDivElement

  constructor (comment: Comment) {
    this.comment = comment
    this.#createCommentBox()
  }

  #createCommentBox (): void {
    this.commentBox = document.createElement('div')
    this.commentBox.setAttribute('id', `comment-${this.comment.id}`)
    this.commentBox.setAttribute('class', 'comment-box')
    this.commentBox.style.backgroundColor = this.comment.backgroundColorHex

    this.commentBox.appendChild(this.#createSpanElementWithText())
  }

  #createSpanElementWithText (): HTMLSpanElement {
    const span = document.createElement('span')
    span.innerText = this.comment.text
    return span
  }
}

export interface Comment {
  id: number
  text: string
  backgroundColorHex: string
}

File: TestTask3/Models/Comment.cs
namespace TestTask3.Models
{
    public class Comment
    {
        public int Id { get; set; }

        public string Text { get; set; } = "";

        public string BackgroundColorHex { get; set; } = "#F";

        public string TextColorHex { get; set; } = "#0";


        public int StickyNoteId { get; set; }
    }
}
